packages/x-charts/src/BarChart/BarChart.tsx

```diff
@@ -242,6 +242,10 @@ BarChart.propTypes = {
     right: PropTypes.number,
     top: PropTypes.number,
   }),
+  /**
+   * The maximum size of the bar element.
+   */
+  maxBarSize: PropTypes.number,
```

Comment:
This prop should be inside the `xAxis` config, or it would be the case to move `categoryGapRatio` and `barGapRatio` outside the axis config as well @alexfauquette

Reply:
I also think it would be better for developer to have everything regrouped but at the same time `maxBarSize ` affect the style of the bar like `borderRadius` which is at the same level while `barGapRatio` and `categoryGapRatio` is more about the Axis.
Should I still move it to Axis ?

Reply:
I view these properties as "styling", and it is unlikely that I would want different styling for different axis 🤔 

Is there an advantage of having `barGapRatio` and `categoryGapRatio` inside the axis config? What would happen if I have two axis configs with different values for eg: top/bottom axis? 

Only the "main" axis values are taken into account https://codesandbox.io/p/sandbox/optimistic-rumple-2l94qw?file=%2Fsrc%2FDemo.tsx%3A3%2C1

To me that makes the config a part of the plot, and not the axis

> btw, the type definition for `barGapRatio` and `categoryGapRatio` seems broken, it is not working for me locally as well, unrelated to this PR.

Reply:
I've updated to have the `maxBarSize` on the axis so if needed you can revert it or accept this depending of your decisions